core/templates/domain/topic/topic-update.service.ts

```diff
@@ -22,719 +22,743 @@
+    this._applyTopicPropertyChange(
+      topic, TopicDomainConstants.TOPIC_PROPERTY_URL_FRAGMENT,
+      urlFragment, oldUrlFragment,
+      (changeDict, topic) => {
+        // ---- Apply ----
+        let newUrlFragment = (
+          this._getNewPropertyValueFromChangeDict(changeDict));
+        topic.setUrlFragment(newUrlFragment);
+      }, (changeDict, topic) => {
+        // ---- Undo ----
+        topic.setUrlFragment(oldUrlFragment);
+      });
+  }
+  /**
+   * Changes the thumbnail filename of a topic and records the change in the
+   * undo/redo service.
+   */
+  setTopicThumbnailFilename(
+      topic: Topic, thumbnailFilename: string): void {
+    let oldThumbnailFilename = angular.copy(topic.getThumbnailFilename());
+    this._applyTopicPropertyChange(
+      topic, TopicDomainConstants.TOPIC_PROPERTY_THUMBNAIL_FILENAME,
+      thumbnailFilename, oldThumbnailFilename,
+      (changeDict, topic) => {
+        // ---- Apply ----
+        let thumbnailFilename = (
+          this._getNewPropertyValueFromChangeDict(changeDict));
+        topic.setThumbnailFilename(thumbnailFilename);
+      }, (changeDict, topic) => {
+        // ---- Undo ----
+        topic.setThumbnailFilename(oldThumbnailFilename);
+      });
+  }
+
+  /**
+   * Changes the thumbnail background color of a topic and records the
+   * change in the undo/redo service.
+   */
+  setTopicThumbnailBgColor(
+      topic: Topic, thumbnailBgColor: string): void {
+    let oldThumbnailBgColor = angular.copy(topic.getThumbnailBgColor());
+    this._applyTopicPropertyChange(
+      topic, TopicDomainConstants.TOPIC_PROPERTY_THUMBNAIL_BG_COLOR,
+      thumbnailBgColor, oldThumbnailBgColor,
+      (changeDict, topic) => {
+        // ---- Apply ----
+        let thumbnailBgColor = (
+          this._getNewPropertyValueFromChangeDict(changeDict));
+        topic.setThumbnailBgColor(thumbnailBgColor);
+      }, (changeDict, topic) => {
+        // ---- Undo ----
+        topic.setThumbnailBgColor(oldThumbnailBgColor);
+      });
+  }
+
+  /**
+   * Changes the description of a topic and records the change in the
+   * undo/redo service.
+   */
+  setTopicDescription(
+      topic: Topic, description: string): void {
+    let oldDescription = angular.copy(topic.getDescription());
+    this._applyTopicPropertyChange(
+      topic, TopicDomainConstants.TOPIC_PROPERTY_DESCRIPTION,
+      description, oldDescription,
+      (changeDict, topic) => {
+        // ---- Apply ----
+        var description = this._getNewPropertyValueFromChangeDict(changeDict);
+        topic.setDescription(description);
+      }, (changeDict, topic) => {
+        // ---- Undo ----
+        topic.setDescription(oldDescription);
+      });
+  }
+
+  /**
+   * Changes the language code of a topic and records the change in
+   * the undo/redo service.
+   */
+  setTopicLanguageCode(
+      topic: Topic, languageCode: string): void {
+    let oldLanguageCode = angular.copy(topic.getLanguageCode());
```

Comment:
Don't use the `angular` variable in migrated code. In this case `deepClone` from lodash should work fine. Here and everywhere else.

Reply:
I have updated with `cloneDeep`.